api/test/trace/propagation/http_text_format_test.cc

```diff
@@ -0,0 +1,206 @@
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/trace/tracer.h"
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/propagation/http_text_format.h"
+#include "opentelemetry/trace/propagation/http_trace_context.h"
+
+using namespace opentelemetry;
+
+static nostd::string_view Getter(const std::map<std::string, std::string> &carrier,
+                                 nostd::string_view trace_type = "traceparent")
+{
+  auto it = carrier.find(std::string(trace_type));
+  if (it != carrier.end())
+  {
+    return nostd::string_view(it->second);
+  }
+  return "";
+}
+
+static void Setter(std::map<std::string, std::string> &carrier,
+                   nostd::string_view trace_type        = "traceparent",
+                   nostd::string_view trace_description = "")
+{
+  carrier[std::string(trace_type)] = std::string(trace_description);
+}
+
+static trace::propagation::HttpTraceContext<std::map<std::string, std::string>> format =
+    trace::propagation::HttpTraceContext<std::map<std::string, std::string>>();
+
+static const nostd::string_view trace_id = "12345678901234567890123456789012";
+static const nostd::string_view span_id  = "1234567890123456";
+
+using MapHttpTraceContext =
+    trace::propagation::HttpTraceContext<std::map<std::string, std::string>>;
+TEST(HTTPTextFormatTest, TraceIdBufferGeneration)
+{
+  constexpr uint8_t buf[] = {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+  EXPECT_EQ(MapHttpTraceContext::GenerateTraceIdFromString("01020304050607080807aabbccddeeff"),
+            trace::TraceId(buf));
+}
+
+TEST(HTTPTextFormatTest, SpanIdBufferGeneration)
+{
+  constexpr uint8_t buf[] = {1, 2, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+  EXPECT_EQ(MapHttpTraceContext::GenerateSpanIdFromString("0102aabbccddeeff"), trace::SpanId(buf));
+}
+
+TEST(HTTPTextFormatTest, TraceFlagsBufferGeneration)
+{
+  EXPECT_EQ(MapHttpTraceContext::GenerateTraceFlagsFromString("00"), trace::TraceFlags());
+}
+
+// TEST(HTTPTextFormatTest, HeadersWithTraceState)
```

Comment:
Please cleanup the commented code. It's not clear if it's meant to be included or not.

Reply:
The commented out code was because I had to comment out all codes related to TraceState and I would need to restore it in the future once TraceState Implementation is done

Reply:
As an alternative I have added additional comments above these commented out codes to explain why they are commented out and should be uncommented once TraceState comes out